src/transform.jl

```diff
@@ -14,11 +14,17 @@ immutable AffineTransform{T,N}
 end
 function AffineTransform(S, offset)
     T = promote_type(eltype(S),eltype(offset))
```

Comment:
Let's change this to `promote(eltype(S), eltype(inv(S)))`. We'd want to calculate `inv(S)` only once, so it would be good to change the inner constructor to simply accept it as an argument.